internal/exp/metrics/metrics.go

```diff
@@ -0,0 +1,120 @@
+package metrics
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/streams"
+)
+
+// MergeMetrics will merge the metrics data in mdB into mdA, then return mdA.
+// mdB will not be modified. The function will attempt to merge the data in mdB into
+// existing ResourceMetrics / ScopeMetrics / Metrics in mdA if possible. If they don't
+// exist, new entries will be created as needed.
+//
+// NOTE: Any "unnecessary" duplicate entries in mdA will *not* be combined. For example if
+// mdA contains two ResourcMetric entries with identical Resource values, they will not be
+// combined. If you wish to have this behavior, you could call this function twice:
+//
+//	cleanedMetrics := MergeMetrics(pmetric.NewMetrics(), mdA)
+//	MergeMetrics(cleanedMetrics, mdB)
+//
+// That said, this will do a large amount of memory copying
+func MergeMetrics(mdA pmetric.Metrics, mdB pmetric.Metrics) pmetric.Metrics {
```

Comment:
```suggestion
func Merge(mdA pmetric.Metrics, mdB pmetric.Metrics) pmetric.Metrics {
```

its already in the package name: `metrics.Merge`, also Go style guides prefer not to stutter: https://sben.dev/stutter/